framework_tvb/tvb/interfaces/web/controllers/simulator_controller.py

```diff
@@ -838,6 +854,22 @@ def set_monitor_params(self, current_monitor, **data):
                 SimulatorWizzardURLs.SET_MONITOR_PARAMS_URL, current_monitor)
             return rendering_rules.to_dict()
 
+        if isinstance(monitor, Projection) and cherrypy.request.method == 'POST':
+            # load region mapping
+            region_mapping_index = ABCAdapter.load_entity_by_gid(data['region_mapping'])
+            region_mapping = h5.load_from_index(region_mapping_index)
+            monitor.region_mapping = region_mapping.gid
```

Comment:
This is really strange anyway .... why load the full HasTraits object in memory to only get the GID from it?
What am I missing ?

Reply:
You are right, these are not necessary since we already have the correct GIDs as strings in `data`